ui/src/table.test.tsx

```diff
@@ -630,7 +630,21 @@ describe('Table.tsx', () => {
       expect(groupHeaders[2]).toHaveTextContent('2012-09-14T18:26:01(1)')
     })
 
-    it('Sorts grouped list', () => {
+    it('Checks if groups contain rows after sort is applied', () => {
+      const { container, getAllByText, getByTestId, getAllByRole } = render(<XTable model={tableProps} />)
+
+      fireEvent.click(getByTestId('groupby'))
+      fireEvent.click(getAllByText('Col1')[1]!)
+      fireEvent.click(container.querySelector('.ms-DetailsHeader-collapseButton')!)
+
+      expect(getAllByRole('gridcell')[3].textContent).toBe(cell21)
+
+      fireEvent.click(container.querySelector('.ms-DetailsHeader-cellTitle i[class*=sortingIcon]')!)
+
+      expect(getAllByRole('gridcell')[3].textContent).toBe(cell21)
```

Comment:
What's the point of this test? It checks if `cell21` is present at a particular place and then it checks it didn't change after sorting? I would assume we want to test whether the sort actually changed row position.

Reply:
The point was to check the correct viewing of rows in case we have as much groups created as possible using groupby. So to group it by column containing unique value in each row. This was the most common case where our group sorting did not work. As far as there is just one value in each group in this case, I can expect it will stay the same after applying sort. Before the group sorting fix there was no rows shown in the group and this test was failing.

Reply:
It would be better to test sort on > 1 row and check if the sort actually worked.

Reply:
Test case whether sort works or not is already implemented: https://github.com/h2oai/wave/blob/56a9ce8f2ef3a48bc3a956aa178e16de1aaf3acc/ui/src/table.test.tsx#L647 

However this test was working even before the sorting issue fix.
Is this what you mean?

Reply:
Yes, I just want a test that makes sure grouped by rows are correctly sorted after sorting.

>However this test was working even before the sorting issue fix.

How is that possible?

Reply:
Let's make following test case mentioned above:
```ts
it('Sorts rows inside the group of the grouped list', () => {
   const { container, getAllByText, getByTestId, getAllByRole } = render(<XTable model={tableProps} />)

   fireEvent.click(getByTestId('groupby'))
   fireEvent.click(getAllByText('Col2')[1]!)
   fireEvent.click(container.querySelector('.ms-DetailsHeader-collapseButton')!)

   expect(getAllByRole('gridcell')[3].textContent).toBe(cell11)

   fireEvent.click(container.querySelector('.ms-DetailsHeader-cellTitle i[class*=sortingIcon]')!)

   expect(getAllByRole('gridcell')[3].textContent).toBe(cell21)
})
```

on following data:
```ts
rows: [
   { name: 'rowname1', cells: ['Quick brown fox.', 'Group1'] },
   { name: 'rowname2', cells: ['Jumps over a dog.', 'Group1'] },
   { name: 'rowname3', cells: ['Wooo hooo.', 'Group2'] }
]
```

We are grouping by `Col2` so now we have two groups `Group1` and `Group2` and its alphabetical order is the same as default one. In this case sorting works as expected and this test case does not reflect our issue.

I can replace name Group1 with Group2 and vice-versa for this test case to reflect also our issue and remove the test case we are commenting.

Reply:
I've edited the existing test case to reflect our issue and now it passes only after the issue is fixed:
```ts
    it('Sorts rows inside the group of the grouped list', () => {
      // test with non-alphabetical groups order to check whether group alphabetical sorting on groupby is considered
      tableProps = {
        ...tableProps,
        rows: [
          { name: 'rowname1', cells: [cell11, 'Group2'] },
          { name: 'rowname2', cells: [cell21, 'Group2'] },
          { name: 'rowname3', cells: [cell31, 'Group1'] }
        ]
      }
      const { container, getAllByText, getByTestId, getAllByRole } = render(<XTable model={tableProps} />)

      fireEvent.click(getByTestId('groupby'))
      fireEvent.click(getAllByText('Col2')[1]!)
      fireEvent.click(container.querySelector('.ms-DetailsHeader-collapseButton')!)

      expect(getAllByRole('gridcell')[8].textContent).toBe(cell11)

      fireEvent.click(container.querySelector('.ms-DetailsHeader-cellTitle i[class*=sortingIcon]')!)

      expect(getAllByRole('gridcell')[8].textContent).toBe(cell21)
    })
```

If you are curious why `getAllByRole('gridcell')[8]`, here is the return value of `getAllByRole('gridcell').map(item => item.innerHTML)` from debug console after groupby and before sort:
```
0:'<button data-is-focusable="false" type="button" class="ms-GroupHeader-expand expand-269" aria-expanded="true" aria-label="expand collapse group"><i data-icon-name="ChevronRightMed" aria-hidden="true" class="expandIsCollapsed-278"></i></button>'
1:'<span>Group1</span><span class="headerCount-267">(1)</span>'
2:''
3:'<button type="button" class="ms-Link root-169" tabindex="-1">Wooo hooo.</button>'
4:'Group1'
5:'<button data-is-focusable="false" type="button" class="ms-GroupHeader-expand expand-269" aria-expanded="true" aria-label="expand collapse group"><i data-icon-name="ChevronRightMed" aria-hidden="true" class="expandIsCollapsed-278"></i></button>'
6:'<span>Group2</span><span class="headerCount-267">(2)</span>'
7:''
8:'<button type="button" class="ms-Link root-169" tabindex="-1">Quick brown fox.</button>'
9:'Group2'
10:''
11:'<button type="button" class="ms-Link root-169" tabindex="-1">Jumps over a dog.</button>'
12:'Group2'
```